pallets/utxo/src/authorship.rs

```diff
@@ -0,0 +1,55 @@
+// Copyright (c) 2021 RBB S.r.l
+// opensource@mintlayer.org
+// SPDX-License-Identifier: MIT
+// Licensed under the MIT License;
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+// 	http://spdx.org/licenses/MIT
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+//
+// Author(s): C. Yap
+
+use crate::{BlockAuthor, Config};
+use frame_support::traits::ValidatorSet;
+use frame_support::{traits::FindAuthor, ConsensusEngineId};
+
+/// Get the `H256` version of the BlockAuthor.
+/// The BlockAuthor is known by its index position only.
+/// To return the actual accountId, the Validators set is required.
+pub struct FindAccountFromAuthorIndex<T, Inner, Validators>(
+    core::marker::PhantomData<(T, Inner, Validators)>,
+);
+
+/// The inner `FindAuthor` will point to the BlockAuthor's index position, given the digest.
+/// That index position points to an index of a list of validators, which this pallet doesn't have.
+/// The `Validators` is a list that can be used to retrieve the account Id.
+impl<T: Config, Inner: FindAuthor<u32>, Validators: ValidatorSet<T::AccountId>>
+    FindAuthor<Validators::ValidatorId> for FindAccountFromAuthorIndex<T, Inner, Validators>
+{
+    fn find_author<'a, I>(digests: I) -> Option<Validators::ValidatorId>
```

Comment:
Can you please explain why we're implementing this ourselves? If I understand correctly, Aura has this already implemented. Why does a digest have to be involved? This sounds more like BABE way of doing things.

Reply:
I wanted to store the "block author" in the utxo side.

Yes, I'm utilizing Aura to identify the block author. I pass the digest to the **_`Inner`_** `FindAuthor`, which is Aura.


https://github.com/mintlayer/core/pull/83/files#diff-0ec06ea58bd455f09ce6b3bb4c2c1c0d37bda51c1e1be2151c560c9c973959ecR482

Reply:
update: I've removed this file.